### DISCUSSION TITLE
/adding-managed-property-mappings-display-templates-may-not-altered/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#111] [Comments]
Adding managed property mappings to your display templates which may not be altered
        
[https://www.eliostruyf.com/adding-managed-property-mappings-display-templates-may-not-altered/](https://www.eliostruyf.com/adding-managed-property-mappings-display-templates-may-not-altered/)

**Comment by comment-migrator** [comment]
```header
Originally posted by Samul on 2015-03-10 21:54
```

Hi Elio,

Very nice and informative articles on display templates..Nice work.

I have a requirement to pull all the documents that are either created or modified by current log in on a site collection along with the total files size of all the documents displayed in CSWP.

Am able to pull the documents by querying with the Author and Modified By managed properties, but am unable to get the sum of all the file sizes.

In the item display template am able to retrieve the individual file size of the documents but I need to sum them up all together to get the total value. Is there any way we can access the file size of the documents in Control display template where I can loop through the items and sum up the total file size?

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2015-03-13 16:25
```

Samul,

You could also do it from within the item display template. You need to create a new object in which you store the total file sizes. For each result that gets rendered, you add the current document file size to the total file size object.

Sample code looks like this:

<br>    var totalSize = 0;<br><br>    function setTotalFileSize (currentSize) {<br>        totalSize = totalSize + currentSize.value;<br>        return Srch.U.toFileSizeDisplay(Math.floor(totalSize), true);<br>    }<br><br>    Srch.U.registerRenderTemplateByName("settotalfilesize", setTotalFileSize);<br>  
  

  
  
    

\_#= size =#\_

  

Regards,  
Elio

**Reply by comment-migrator** [reply]
```header
Originally posted by Domtar SharePoint on 2015-10-26 15:20
```

But how did you get the file size, witch managed property (crawled property) you are using ?

**Reply by comment-migrator** [reply]
```header
Originally posted by Samul on 2015-03-14 00:36
```

Thanks Elio,

I could have done that if I had only one page, but it will not give me total size of all the documents across all the pages.

The above code sums up the size of documents in the current page. I need the sum of sizes of all the search results across the pages.

Regards,  
Samul

**Reply by comment-migrator** [reply]
```header
Originally posted on 2015-10-26 15:22
```

There is a Managed Property called "Size", this will give you the file size of the corresponding document.

**Reply by comment-migrator** [reply]
```header
Originally posted by Elio Struyf on 2015-03-17 01:46
```

That would be quite difficult to do. Depending of the amount of documents, the pages can vary.

An option could be to retrieve the results via a rest calls. When the fist set is retrieved, you calculate the total size of them and check if there are other results to load. But still this will be a lot of rest calls depending on the amount of documents you got in your environment.

**Comment by r3brar** [comment]
Hi Elio,

One query regarding mapping managed property in Search display template. If a value being returned by a property is True/False and if I need to convert it to more user friendly text like Yes/No. Is there a way to do that in Item Display Template. I tried putting some if conditions to validate if managed property value is True, set it to Yes otherwise no but no luck. Any guidance to make this check and replacing managed property value to Yes/No will be very helpful.
